camkes_vm_helpers.cmake

```diff
@@ -63,18 +63,132 @@ function(DeclareCAmkESVM init_component)
     )
 endfunction(DeclareCAmkESVM)
 
+#
 # Function defines a CAmkESVMFileServer using the declared fileserver images
 # and fileserver dependencies. These images are placed into a CPIO archive.
+#
+# Parameters:
+#
+# TYPE <type>
+#   Type of the file server CAmkES component.
+#   Optional, defaults to "FileServer"
+#
+# INSTANCE <name>
+#   Instance name of the file server CAmkES component.
+#   Optional, defaults to "fserv"
+#
+# FILES [<CPIO_NAME>:]<FILE_NAME>[ <CPIO_NAME>:]<FILE_NAME>[...]]
+#  Files to be added to the file server.
+#
+# DEPENDS <dep>[ <dep>[...]]
+#   Any additional dependencies for the file/image the caller is adding to the
+#   file server
+#
+#
 function(DefineCAmkESVMFileServer)
-    # Retrieve defined kernel images, rootfs images and extraction dependencies
-    get_target_property(fileserver_images vm_fserver_config FILES)
-    get_target_property(fileserver_deps vm_fserver_config DEPS)
-    # Build CPIO archive given the defined kernel and rootfs images
+
+    cmake_parse_arguments(
+        PARSE_ARGV
+        0
+        PARAM # variable prefix
+        "" # option arguments
+        "TYPE;INSTANCE" # optional single value arguments
+        "FILES;DEPENDS" # optional multi value arguments
+    )
+
+    if(PARAM_UNPARSED_ARGUMENTS)
+        message(FATAL_ERROR "Unknown arguments: ${PARAM_UNPARSED_ARGUMENTS}")
+    endif()
+
+    if(NOT PARAM_TYPE)
+        set(PARAM_TYPE "FileServer")
+    endif()
+
+    if(NOT PARAM_INSTANCE)
+        set(PARAM_INSTANCE "fserv")
+    endif()
+
+    if(PARAM_DEPENDS)
+        set_property(
+            TARGET vm_fserver_config
+            APPEND
+            PROPERTY DEPS_${PARAM_INSTANCE} ${PARAM_DEPENDS}
+        )
+    endif()
+
+    foreach(item IN LISTS PARAM_FILES) # [<CPIO_NAME>:]<FILE_NAME>
+        string(
+            REPLACE
+                ":"
+                ";"
+                item_as_list
+                "${item}"
+        )
+        list(LENGTH item_as_list len)
+        if(len EQUAL 2)
+            list(POP_FRONT item_as_list CPIO_NAME)
+            list(POP_FRONT item_as_list FILE_NAME)
+        elseif(len EQUAL 1)
+            list(POP_FRONT item_as_list FILE_NAME)
+            get_filename_component(CPIO_NAME "${FILE_NAME}" NAME)
+        else()
+            message(FATAL_ERROR "invalid format: ${item}")
+        endif()
+        # No need to pass dependencies here, we have already added them above.
+        AddToFileServer("${CPIO_NAME}" "${FILE_NAME}" INSTANCE "${PARAM_INSTANCE}")
+    endforeach()
+
+    # now process the file/deps list
+    get_target_property(files vm_fserver_config FILES_${PARAM_INSTANCE})
+    get_target_property(deps vm_fserver_config DEPS_${PARAM_INSTANCE})
```

Comment:
If the target property `DEPS_${PARAM_INSTANCE}` does not exist (because there are no dependencies), this will set `deps` to `deps-NOTFOUND` and then we see built errors lile `ninja: error: 'apps/Arm/vm_minimal/deps-NOTFOUND', needed by 'apps/Arm/vm_minimal/fserv/files/linux', missing and no known rule to make it`. Check for that case t avoid the error.

Reply:
Added a check for this cases and handle it. The case where we have no dependencies (and/or no files) works now as before.. There is no warning/error if there are no files, as this still creates a valid CPIO archive and it might be a valid use case in rare corner cases.